### PR TITLE
CSHARP-4733: Add implementation for Atlas Search Tracking search terms

### PR DESCRIPTION
**Pull request by sboulema**
The 7 June 2023 release of MongoDB Atlas Search introduced support for tracking search terms.

[07 June 2023 Release](https://www.mongodb.com/docs/atlas/atlas-search/changelog/#07-june-2023-release)

[Track search terms](https://www.mongodb.com/docs/atlas/atlas-search/tracking/#std-label-fts-tracking-ref)

It would be very nice to have access to the tracking field from the C# driver

### CODE REVIEW

**Comment by sboulema** [comment]
Update: Use the new SearchOptions overload instead

**Comment by BorisDog** [comment]
Somehow this PR shows unrelated changes, and contains commits that are already in master.
Can you please rebase this PR on latest master, so it contains only relevant commits to this change?

Thanks.

**Comment by sboulema** [comment]
PR cleaned :)

**Review by sanych-sun — APPROVED** [review]
LGTM

File: tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs [hunk 1]
```diff
@@ -194,6 +194,25 @@ public void Search_should_add_expected_stage_with_score_details()
             stages[0].Should().Be("{ $search: { text: { query: 'foo', path: 'bar' }, scoreDetails: true } }");
         }
 
+        [Fact]
+        public void Search_should_add_expected_stage_with_tracking()
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs:198]: Place this method in alphabetical order.

**Review by BorisDog — CHANGES_REQUESTED** [review]

File: tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs (hunk 1 above)
Comment (sboulema) [reply tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs:211]: Done :)

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**